text.py

```diff
@@ -315,16 +437,93 @@ def __init__(self,
+                dtype=self._dtype)
+
+        if "h" in update_gate_weights and update_gate_weights["h"] is not None:
+            self.ug_h = update_gate_weights["h"]
+        else:
+            self.ug_h = self.create_parameter(
+                attr=gate_param_attr,
+                shape=[self._hidden_size, self._hidden_size],
+                dtype=self._dtype)
+
+        if "b" in update_gate_weights and update_gate_weights["b"] is not None:
+            self.ug_b = update_gate_weights["b"]
+        else:
+            self.ug_b = self.create_parameter(
+                attr=gate_bias_attr,
+                shape=[self._hidden_size],
+                dtype=self._dtype,
+                is_bias=True)
+
+        # reset gate parameters
+        if "w" in reset_gate_weights and reset_gate_weights["w"] is not None:
+            self.rg_w = reset_gate_weights["w"]
+        else:
+            self.rg_w = self.create_parameter(
+                attr=gate_param_attr,
+                shape=[self._input_size, self._hidden_size],
+                dtype=self._dtype)
+
+        if "h" in reset_gate_weights and reset_gate_weights["h"] is not None:
+            self.rg_h = reset_gate_weights["h"]
+        else:
+            self.rg_h = self.create_parameter(
+                attr=gate_param_attr,
+                shape=[self._hidden_size, self._hidden_size],
+                dtype=self._dtype)
+
+        if "b" in reset_gate_weights and reset_gate_weights["b"] is not None:
+            self.rg_b = reused_params["b"]
+        else:
+            self.rg_b = self.create_parameter(
+                attr=gate_param_attr,
+                shape=[self._hidden_size],
+                dtype=self._dtype,
+                is_bias=True)
+
+        # cell parameters
+        if "w" in cell_weights and cell_weights["w"] is not None:
+            self.c_w = cell_weights["w"]
+        else:
+            self.c_w = self.create_parameter(
+                attr=candidate_param_attr,
+                shape=[self._input_size, self._hidden_size],
+                dtype=self._dtype)
+
+        if "h" in cell_weights and cell_weights["h"] is not None:
+            self.c_h = cell_weights["h"]
+        else:
+            self.c_h = self.create_parameter(
+                attr=candidate_param_attr,
+                shape=[self._hidden_size, self._hidden_size],
+                dtype=self._dtype)
+
+        if "b" in cell_weights and cell_weights["b"] is not None:
+            self.c_b = cell_weights["b"]
+        else:
+            self.c_b = self.create_parameter(
+                attr=candidate_bias_attr,
+                shape=[self._hidden_size],
+                dtype=self._dtype,
+                is_bias=True)
+
+        rg_weights = layers.concat([self.rg_w, self.rg_h], axis=-1)
+        ug_weights = layers.concat([self.ug_w, self.ug_h], axis=-1)
+        self._gate_weight = layers.concat([rg_weights, ug_weights], axis=0)
```

Comment:
Should we concat `_w` and `_h` by `axis=0`, and `rg_weights, ug_weights` by `axis=-1`